broker/pkg/coordinator/coordinator.go

```diff
@@ -93,10 +93,23 @@ func (c *Coordinator) AttachToServer(server *GRPCServer) {
 		ConsumerHandshake:    c.consumerHandshake,
 		ConsumerConnected:    c.consumerConnected,
 		ConsumerDisconnected: c.consumerDisconnected,
-		DeleteMessagesListener: func(ctx context.Context, timelineID string, msgs []timeline.Message) []errors.MessageIDTuple {
-			//TODO add here a way to identify the consumer or producer
-			//only specific clients can delete specific messages
-			return c.storage.Delete(ctx, []byte(timelineID), msgs)
+		DeleteMessagesListener: func(ctx context.Context, sessionID string, timelineID string, msgs []timeline.Message) []errors.MessageIDTuple {
+			consumer, err := c.greeter.GetConsumer(sessionID)
+			if err == nil {
+				for i := range msgs {
+					msgs[i].LockConsumerID = consumer.ID
```

Comment:
It does not work like this. This field is used to determine the Consumer that has the Lease. 
Consumer.ID here is the consumer that makes the delete request. We cannot guarantee that they are the same consumer. 
We can add  new fields in the Delete() signature to indicate which Consumer or Producer is making the request, and Storage can check if its message or not.